addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -285,7 +285,8 @@ const Wysiwyg = Widget.extend({
         this.$editable.on('click', '.o_image, .media_iframe_video', e => e.preventDefault());
         this.showTooltip = true;
         this.$editable.on('dblclick', mediaSelector, function () {
-            if (this.isContentEditable || (this.parentElement && this.parentElement.isContentEditable)) {
+            if (this.isContentEditable || (this.parentElement &&
+                (this.parentElement.isContentEditable || this.parentElement.classList.contains('o_grid_item_image')))) {
```

Comment:
@Zinston @dmo-odoo Following @rdeodoo's ask, I don't really understand your original diff (why is not there any comment next to this condition?). "if I have an image which is not editable but their parent is, the image is in fact editable"? What case is that ?

@sobo-odoo In any case, not a fan of your fix, spreading even more grid code in the codebase. This part here in wysiwyg.js should be as generic as possible, we cannot make (keep making?) exceptions like this. Seeing the condition, we could make it work by automatically adding `contenteditable="true"` on those grid images when entering edit mode I think? (around the same code that automatically add `contenteditable="false"` on the parent?). That would actually make sense with or without this condition here.

Reply:
@qsm-odoo `isContentEditable` doesn't determine whether the _element itself_ is editable but rather whether its _contents_ are editable (see [mdn](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/isContentEditable)). In the case of an image it doesn't mean much since `img` is a self-closing element. Granted it's a weird case that this covers, where somehow an image was marked as `contenteditable="false"` for some reason. But IIRC this happens in `Knowledge`.

Reply:
Yep, I agree, strange case in general and the spec is probably not clear about `contenteditable` on self closing tags. And even if it was clear, it would probably be implemented differently by browsers as always. That's actually the case::
```html
<div contenteditable="true">
    Hello <img contenteditable="false" src="..."/> world
</div>
```
Chrome: `contenteditable="false"` on the image makes it so backspace at the start of "world" does not delete the image (but not delete at the end of "Hello" :shrug:)
Firefox: No effect, the image can be deleted

 -> the more the reason to add comments to explain the usecase IMO :grimacing:

But in the case of our editor, I think that this configuration should mean that the image is not editable via the interface (why would the developer add `contenteditable="false"` otherwise ?). And if we think that we should not care about it (debatable)... then we really should not care about it and not add exception in the code to support the usecase IMO :shrug: If this happens in Knowledge, let's stop making it happen in Knowledge :wink: Or again add comments why we cannot fix Knowledge :wink:

In our usecase here, we actually want to tell the opposite: do not allow to edit the image's parent text content but allow to edit the image. So this case:
```html
<div contenteditable="false">
    <img src="..."/> <!-- I suggest adding contenteditable="true" here -->
</div>
```
Also a strange case that `contenteditable` alone probably does not specify (there is no HTML equivalent to have a container with an image where only the image is editable).
@sobo-odoo Again, I think that adding `contenteditable="true"` on the image would make sense independently of the editor and/or HTML spec (and would fix our issue for the moment :+1:). Could be a good occasion to start making tests about the grid mode in general, adding and replacing images could be a nice flow to start :+1: (Trying to test resizing etc could be too difficult for the moment though but just switching to grid and replacing images would be easy to test).